src/node_report.cc

```diff
@@ -0,0 +1,1024 @@
+  json_indent(indent);
+  json_keyvaluenormal(
+      out, *indent, "memory_size", v8_heap_space_stats.space_size());
+  json_keyvaluenormal(out,
+                      *indent,
+                      "committed_memory",
+                      v8_heap_space_stats.physical_space_size());
+  json_keyvaluenormal(out,
+                      *indent,
+                      "capacity",
+                      v8_heap_space_stats.space_used_size() +
+                          v8_heap_space_stats.space_available_size());
+  json_keyvaluenormal(
+      out, *indent, "used", v8_heap_space_stats.space_used_size());
+  json_keyvaluefinal(
+      out, *indent, "available", v8_heap_space_stats.space_available_size());
+  json_deindent(indent);
+  json_objectendfinal(out, *indent);
+  json_deindent(indent);
+  json_objectendfinal(out, *indent);
+  json_deindent(indent);
+  json_objectendnormal(out, *indent);
+}
+
+#ifndef _WIN32
+/*******************************************************************************
+ * Function to print resource usage (Linux/OSX only).
+ *
+ ******************************************************************************/
+static void PrintResourceUsage(std::ostream& out, int* indent) {
+  char buf[64];
+  double cpu_abs;
+  double cpu_percentage;
+  time_t current_time;  // current time absolute
+  time(&current_time);
+  auto uptime = difftime(current_time, boot_time);
+  if (uptime == 0)
+    uptime = 1;  // avoid division by zero.
+
+  // Process and current thread usage statistics
+  struct rusage stats;
+  json_objectstart(out, *indent, "resource_usage");
+  json_indent(indent);
+  if (getrusage(RUSAGE_SELF, &stats) == 0) {
+#if defined(__APPLE__) || defined(_AIX)
+    snprintf(buf, sizeof(buf), "%ld.%06d",
+             stats.ru_utime.tv_sec, stats.ru_utime.tv_usec);
+    json_keyvaluenormal(out, *indent, "user_cpu_seconds", buf);
+    snprintf(buf, sizeof(buf), "%ld.%06d",
+             stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
+    json_keyvaluenormal(out, *indent, "kernel_cpu_seconds", buf);
+#else
+    snprintf(buf, sizeof(buf), "%ld.%06ld",
+             stats.ru_utime.tv_sec, stats.ru_utime.tv_usec);
+    json_keyvaluenormal(out, *indent, "user_cpu_seconds", buf);
+    snprintf(buf, sizeof(buf), "%ld.%06ld",
+             stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
+    json_keyvaluenormal(out, *indent, "kernel_cpu_seconds", buf);
+#endif
+    cpu_abs = stats.ru_utime.tv_sec + 0.000001 * stats.ru_utime.tv_usec +
+              stats.ru_stime.tv_sec + 0.000001 *  stats.ru_stime.tv_usec;
+    cpu_percentage = (cpu_abs / uptime) * 100.0;
+    json_keyvaluenormal(
+        out, *indent, "cpu_consumption_percent", cpu_percentage);
+    // WriteInteger(out, stats.ru_maxrss * 1024);
+    json_keyvaluenormal(out, *indent, "max_rss", stats.ru_maxrss * 1024);
+    snprintf(buf,
+             sizeof(buf),
+             "%ld (I/O required) %ld (no I/O required)",
```

Comment:
Maybe spilt this into two separate values?

Maybe this is just me, but it might be nice to end up with a fully programmatically accessible report that requires no parsing of individual strings…?

Reply:
split into `IORequired` and `IONotRequired` under a top level object called `fsActivity`

agree that JSON data should be readily consumable for programs without further tailoring